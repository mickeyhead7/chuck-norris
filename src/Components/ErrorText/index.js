import React from 'react';
import './style.css';

const ErrorText = ({ error }) => error ? (
  <div className="error">
    {error}
  </div>
) : null;

export default ErrorText;
