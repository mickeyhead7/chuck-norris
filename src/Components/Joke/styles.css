.joke {
  background-color: lightgrey;
  margin: 0 0 1rem;
  padding: 1rem;
}
