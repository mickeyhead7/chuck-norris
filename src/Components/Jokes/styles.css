.jokes {
  margin: 1rem;
}
