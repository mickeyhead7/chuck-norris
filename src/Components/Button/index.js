import React from 'react';
import './styles.css';

const Button = ({ disabled, onClick }) => (
  <button
    className="button"
    disabled={disabled}
    onClick={onClick}
  >
    Get a new joke
  </button>
);

export default Button;
