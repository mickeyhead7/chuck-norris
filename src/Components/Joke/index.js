import React from 'react';
import './styles.css';

const Joke = ({ id, joke }) => (
  <div className="joke">
    {joke}
  </div>
);

export default Joke;
