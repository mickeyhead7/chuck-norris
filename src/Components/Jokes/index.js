import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchJoke } from '../../store/actions/jokes';
import { errorSelector, jokesSelector, inProgressSelector } from '../../store/reducers/jokes';
import ErrorText from '../ErrorText';
import Joke from '../Joke';
import Button from '../Button';
import './styles.css';

export const Jokes = ({ error, jokes, fetchJoke, buttonDisabled }) => (
  <div className="jokes">
    <ErrorText error={error} />
    {jokes.map(joke => (
      <Joke key={joke.id} {...joke} />
    ))}
    <Button disabled={buttonDisabled} onClick={fetchJoke} />
  </div>
);

Jokes.propTypes = {
  error: PropTypes.string,
  jokes: PropTypes.array.isRequired,
  fetchJoke: PropTypes.func.isRequired,
  buttonDisabled: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  error: errorSelector(state),
  jokes: jokesSelector(state),
  buttonDisabled: !!inProgressSelector(state)
});

const mapDispatchToProps = dispatch => ({
  fetchJoke: joke => dispatch(fetchJoke(joke))
});

export default connect(mapStateToProps, mapDispatchToProps)(Jokes);
