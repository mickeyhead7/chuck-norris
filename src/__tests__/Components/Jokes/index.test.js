import React from 'react';
import { shallow } from 'enzyme';
import { Jokes } from '../../../Components/Jokes';
import { Joke } from '../../../Components/Joke';

describe('Jokes', () => {
  test('Jokes default', () => {
    const props = {
      error: null,
      jokes: [
        {
          id: 1,
          joke: 'foo'
        },
        {
          id: 2,
          joke: 'bar'
        }
      ],
      fetchJoke: jest.fn(),
      buttonDisabled: false
    };
    const jokes = shallow(<Jokes {...props} />);
    const joke = jokes.find('Joke');

    expect(jokes).toMatchSnapshot();
    expect(joke).toHaveLength(2);
  });
});