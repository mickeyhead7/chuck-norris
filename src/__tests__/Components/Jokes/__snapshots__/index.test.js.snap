// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jokes Jokes default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    buttonDisabled={false}
    error={null}
    fetchJoke={[Function]}
    jokes={
        Array [
            Object {
              "id": 1,
              "joke": "foo",
            },
            Object {
              "id": 2,
              "joke": "bar",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ErrorText
          error={null}
/>,
        Array [
          <Joke
            id={1}
            joke="foo"
/>,
          <Joke
            id={2}
            joke="bar"
/>,
        ],
        <Button
          disabled={false}
          onClick={[Function]}
/>,
      ],
      "className": "jokes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "id": 1,
          "joke": "foo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "id": 2,
          "joke": "bar",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ErrorText
            error={null}
/>,
          Array [
            <Joke
              id={1}
              joke="foo"
/>,
            <Joke
              id={2}
              joke="bar"
/>,
          ],
          <Button
            disabled={false}
            onClick={[Function]}
/>,
        ],
        "className": "jokes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "id": 1,
            "joke": "foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "id": 2,
            "joke": "bar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
